### ISSUE TITLE
Not able to access Grafana UI until auto provisioning is done

### ISSUE THREAD

**Issue by vijayakumarbathini** [grafana/grafana#21133]
**What happened**:
We are not able to login into Grafana until auto provisioning is completed. We are running grafana through docker image and provisioning around 300 orgs with 50 dashboards each. It is taking 15-20 mins for auto provision to be completed

**What you expected to happen**:
At least able to login into main org or able to see some message saying provisioning is still in progress.
**How to reproduce it (as minimally and precisely as possible)**:
Try to do provisioning with more dashboards and orgs and in parallel try to access grafana UI on 3000 port
**Anything else we need to know?**:

**Environment**:
- Grafana version: 6.3
- Data source type & version: Postgres and 11
- OS Grafana is installed on: Linux
- User OS & Browser: OSx and Chrome
- Grafana plugins:
- Others:

**Comment by yesoreyeram** [comment]
We are having same issue. But relatively small. 15 orgs and just 5 to 20 data sources each. No dashboards. But provisioning takes almost 30 to 40 seconds. So evertime restart process took more than a minute. Can't the provisioning happen asynchronous way ?

**Comment by aknuds1** [comment]
Closing this as a duplicate of #21113, since it should be a symptom of the same issue and we envision the same solution. Thanks!

**Comment by marefr** [comment]
Had a quick lock at the code and seems like data sources, dashboards and alert notifiers are provisioned on Init where's before only data sources and alert notifiers was provisioned on Init, see https://github.com/grafana/grafana/commit/42b745a0987fa631718e4ae956537ac346c904b6#diff-a9672f54b84e56c5da85374613c77bd7R57. 

This change was introduced in 6.2-beta1 it seems and could be the reason for slow startup since `Init`  are run for each service synchronously in server.go, see https://github.com/grafana/grafana/blob/master/pkg/cmd/grafana-server/server.go#L105
